src/software/simulation/physics_simulator_robot.h

```diff
@@ -0,0 +1,311 @@
+#pragma once
+
+#include <cinttypes>
+#include <memory>
+
+#include "software/simulation/firmware_object_deleter.h"
+#include "software/simulation/physics/physics_ball.h"
+#include "software/simulation/physics/physics_robot.h"
+#include "software/simulation/simulator_robot.h"
+
+extern "C"
+{
+#include "firmware/app/primitives/primitive_manager.h"
+#include "shared/proto/primitive.nanopb.h"
+}
+
+/**
+ * The PhysicsSimulatorRobot class acts as a wrapper for a PhysicsRobot that deals with
+ * more logic-focused elements for simulation, such as whether or not autokick is enabled.
+ *
+ * All members of this class are intentionally protected or private to force this class
+ * to only be controlled by the SimulatorRobotSingleton.
+ */
+class PhysicsSimulatorRobot : public SimulatorRobot
+{
+   public:
+    /**
+     * Create a new SimulatorRobot given a PhysicsRobot
+     *
+     * @param physics_robot the PhysicsRobot to simulate and control
+     */
+    explicit PhysicsSimulatorRobot(std::weak_ptr<PhysicsRobot> physics_robot);
+    explicit PhysicsSimulatorRobot() = delete;
+
+    /**
+     * Returns the ID of this robot
+     *
+     * @return the ID of this robot
+     */
+    unsigned int getRobotId() override;
```

Comment:
for all overridden function, no need for javadoc comments

Reply:
Removd